src/auth_db/mod.rs

```diff
@@ -45,29 +48,99 @@ impl<'d> Deserialize<'d> for BounceType {
     }
 }
 
-#[derive(Clone, Copy, Debug, Deserialize)]
+impl Serialize for BounceType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let value = match self {
+            BounceType::Hard => 1,
+            BounceType::Soft => 2,
+            BounceType::Complaint => 3,
+        };
+        serializer.serialize_u8(value)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub enum BounceSubtype {
     // Set by the auth db if an input string is not recognised
-    Unmapped = 0,
+    Unmapped,
```

Comment:
If I'd had the sense to write deserialization tests for `BounceSubtype` in #27, I would have realised then that my derived imlpementation wasn't doing what I thought it was. Hence these changes and explicit implementation of `Deserialize`, which should have been part of that PR.